### PR TITLE
Add Route Structure, required parameters, and url matchings to backend README.md

### LINKED ISSUE #1: Formalize Schema

### PR DESCRIPTION
**Pull request by SAbbas2018**
## [DEV-006] Backend Boilerplate

Fixes: #1 

-   [Y] Has the code been formatted? (N/A)
-   [Y] Have dependencies/packages been added? (N/A)

### Changes
Added to Route structure, URL matching, and possible parameters required in the queries.

### Testing
-   [Y] Does the code have sufficient automated test coverage? If not, include a reasoning and steps required to manually QA (N/A)

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -1,4 +1,60 @@
 # Pizza Backend
```
Comment (jcserv) [inline backend/README.md]: We're going by Dialog!

File: backend/README.md [hunk 2]
```diff
@@ -1,4 +1,60 @@
...
+## Routes
+
+We will require routes for the user to be able to enroll into classes, make posts, reply to posts, etc. These should all be grouped under a user route, where we use either the users name or id
+in the route itself for the ability to "enroll" them into a class.
+
+### Structure
+
+user_router := router.Group("/user"){
+
+    user_router.POST("/register", handleUserRegister) // The request respondst to the url matching "/user/register" and should include
```
Comment (jcserv) [inline backend/README.md]: move this to the next line, it's a bit difficult to read. same for the other routes!

File: backend/README.md [hunk 3]
```diff
@@ -1,4 +1,60 @@
...
+    // information such as first_name, last_name, email. How is user_id being generated? will it be given to us in the request or is is
+    // it made automatically in the database and sent to the front end in a different call
+
+    user_router.POST("/login", handleUserLogin) // The request responds to the url matching "/user/login" and should include
+    // parameters we require for authentication (this will be done later), we can send the user_id to the front end as a response here
+
+    Should we add a reset password for a user route when we add authentication ?
```
Comment (jcserv) [inline backend/README.md]: definitely, good thinking

File: backend/README.md [hunk 4]
```diff
@@ -1,4 +1,60 @@
...
+
+}
+
+The following are all routes for a specific user based on their user_id. Certain routes should only be accessible to professors, e.g. course creation, and archive course.
+
+specific_user_router := router.Group("/user/:user_id"){
```
Comment (jcserv) [inline backend/README.md]: can we do uid?
not as readable but it looks nicer 😎 

similarly, cid for course_id & tid for thread_id

Q: @krishchow 
is golang pothole case or camel case?
the naming for the id variables follows pothole case but in other instances of this file we use camelcase.

File: backend/README.md [hunk 5]
```diff
@@ -1,4 +1,60 @@
...
+    // the request should include information about the post that can be used to find it (thread_id?)
+
+    specific_user_router.POST("/:course_id/:thread_id/createComment", handleComment) // The request responds to the url matching: "/user/:user_id/:course_id/:thread_id/createComment"
+    // Since each comment is specific to a thread in a specific course we will require course_id, and thread_id for comments along with the prerequiste user_id as well
+
+    specific_user_router.DELETE("/:course_id/:thread_id", handleDeletePost) // The request responds to the url matching "/users/:user_id/:course_id/:thread_id/deletePost"
+    // Middleware will be required to make sure only the author of the post, or an instructor is able to delete the post
+    // for author authentication we can check the user_id supplied with the author entry of the thread
+
+    specific_user_router.PUT("/:course_id/:thread_id/:comment_id/upVote", handleUpVote) // The request responds to the url matching "/users/:user_id/:course_id/:thread_id/:comment_id/upVote"
```
Comment (jcserv) [inline backend/README.md]: i think the v in upvote doesn't need to be capitallized

File: backend/README.md [hunk 6]
```diff
@@ -1,4 +1,60 @@
...
+    // This is an update method that allows users to upvote comments on a given post.
+    // PUT can be replaced by PATCH, since both can be used for Updating, but PATCH seems to be for modfiying, maybe use PATCH here instead?
```
Comment (jcserv) [inline backend/README.md]: some typos here:

Updating -> updating
modfiying -> modifying

in regards to the question: i agree that PATCH would be better than PUT based on the definition here:
```
PUT method uses the request URI to supply a modified version of the requested resource which replaces the original version of the resource, whereas the PATCH method supplies a set of instructions to modify the resource.
```

**Review by jcserv — CHANGES_REQUESTED** [review]
Nice work, just a few minor fixes.
Can you also update the README.md in the root directory for the backend project structure?

**Comment by SAbbas2018** [comment]
Fixed all the issues above. I have converted everything to **camelCase**, added the backend structure to the root README.md and added terminology in the backend/README.md.

**Review by jcserv — APPROVED** [review]
Great work on the fixes, I like the shorthand legend

**Review by AipioxTechson — APPROVED** [review]

**Pull request merged**